optimum_benchmark/experiment.py

```diff
@@ -114,13 +113,11 @@ class ExperimentConfig:
 
 @hydra.main(version_base=None)
 def run_experiment(experiment: DictConfig) -> None:
-    # By default, Hydra populates the metadata object_type with the ones from ExperimentConfig but the object_type should really be
-    # one of the subclass (e.g. PyTorchBackendConfig instead of BackendConfig). This is required to call `to_object`.
-    experiment = remap_to_correct_metadata(experiment)
+    from omegaconf import SCMode
 
-    # This is required to trigger __post_init__. Reference: https://github.com/omry/omegaconf/issues/377
-    experiment = OmegaConf.to_object(experiment)
-    experiment = OmegaConf.create(experiment)
+    experiment = OmegaConf.to_container(
+        experiment, structured_config_mode=SCMode.INSTANTIATE
+    )
```

Comment:
`resolve` is False by default for `to_container` while it is True for `to_object`. Does this indeed resolves? Why and where?

I am not sure to remember why I put `OmegaConf.create`, why is it not necessary anymore?

Reply:
yes true, `to_object` is safer and more strict, some things can go unnoticed with `to_container` and stay as strings instead of being resolved

Reply:
I think there's no benefit to using `to_create` because with it we just go back to `DictConfig` objects. 
not sure why we had it there, I thought it was part of the `__post_init__` patch.